### PR TITLE
Cache _get_data_env results for 5 minutes

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- _get_data_environment (which is called every time a query is made) performs a (slow) DB query to look for assets, despite the fact that the information does not change often.

Issue Number: GTC-2779

## What is the new behavior?

- The results of _get_data_environment are cached with a time to live of 5 minutes. This should eliminate a significant portion of queries to the DB without causing much lag in acknowledging new assets.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by solomon-negusse — APPROVED** [review]
Awesome! Just one minor suggestion (maybe for future iteration): what if we clear the cache every time we add a new raster tile set and set `ttl` to a value much larger the 5 mins you set.

**Comment by dmannarino** [comment]
> > Awesome! Just one minor suggestion (maybe for future iteration): what if we clear the cache every time we add a new raster tile set (or set one to latest) and set `ttl` to a value much larger the 5 mins you set.
> 
> Oh, man, that's a great idea. Let me see if I can sneak it in quickly. :)
Actually, I won't right now. I need to think about how to do it right rather than making spaghetti code.
An inclination supported by the fact that my attempt resulted in a circular dependency. ;) So I'll create follow-up ticket and merge this as is. Thanks!

**Review by jterry64 — APPROVED** [review]
Sweet, glad it was just basically a one-liner

File: tests_v2/unit/app/routes/analysis/test_analysis.py [hunk 1]
```diff
@@ -103,6 +106,9 @@ async def test_raster_analysis_payload_shape(
         )
         monkeypatch.setattr(geostore.rw_api, "get_geostore", mock_rw_get_geostore)
 
+        # The other tests will have polluted the data env cache. Clear it.
+        queries._get_data_environment.cache_clear()
```
Comment (gtempus) [inline tests_v2/unit/app/routes/analysis/test_analysis.py:110]: Cool! I learned something new. Nice work, @dmannarino!

**Pull request merged**